### PR TITLE
Define Help Functionality [Closes #66 #67] [Closes PR #85]

### LINKED ISSUE #66: Define Help functionality
As a user, I want to be presented with a help screen when I log in, so that I can easily access information about how to use floworky.
* User should have the option to turn off the help screen so that it is not shown on successive logins
* Help should have a mutable set of topics with screen casts (or text content if easier, for now)
* Help should have a mutable list of keyboard shortcuts

### PR DESCRIPTION
**Pull request by nicosesma**
Did not fix the comment for line 16 in routes/help.js because the line was over 80 characters and could not fit in one line
@jrob8577

### CODE REVIEW

File: routes/accounts.js [hunk 1]
```diff
@@ -4,9 +4,11 @@ const router = express.Router()
 const passport = require( '../auth/passport' )
 const register = require( './accounts/register' )
 const { testForCode, whereClause } = require( './accounts/verify_user' )
+const addUserTopics = require( './accounts/add_user_topics' )
 const validateEmail = require( '../src/mail/validate_email' )
```
Comment (jrob8577) [inline routes/accounts.js]: Only one new line is needed between sections, please remove extra whitespace.

File: routes/accounts.js [hunk 2]
```diff
@@ -17,11 +19,13 @@ router.get( '/register', ( request, response ) => {
...
 router.post( '/register', ( request, response ) => {
-  const { User } = request.app.get( 'models' )
-  const { email, password } =request.body
+  const { User, UserTopic, Topic } = request.app.get( 'models' )
+  const { email, password } = request.body
 
   validateEmail( email )
+    .then( addUserTopics( Topic, UserTopic ) )
     .then( validEmail => register( User, validEmail, password ) )
+
```
Comment (jrob8577) [inline routes/accounts.js]: Remove whitespace

File: routes/help.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+router.post( '/:topicId', ( request, response ) => {
+  const UserTopic = request.app.get('models').UserTopic
+  
+  const { topicId } = request.params
+  const userId = request.user.id
+  const where = { topic_id: topicId, user_id: userId }
+
+  UserTopic.update( { viewed: true }, { where } )
+    .then( result =>
```
Comment (jrob8577) [inline routes/help.js]: Single line please

File: routes/help.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  
+  const { topicId } = request.params
+  const userId = request.user.id
+  const where = { topic_id: topicId, user_id: userId }
+
+  UserTopic.update( { viewed: true }, { where } )
+    .then( result => 
+      response.json({ success: true, topicId })
+    )
+    .catch( error =>
```
Comment (jrob8577) [inline routes/help.js]: Is this the one that doesn't fit on a single line (80 chars)?  If not, please collapse to single line.

File: views/help/howtos.pug [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    .tab-content
+      #howto.tab-pane.fade.in.active
+        h3 How To's
+        iframe(src='https://www.youtube.com/embed/SOPYrxvojVo', frameborder='0')
+        .topic-list
+          ul
+          each topic in topics
+            p
+            li.singleTopic
+              a.btn.btn-info.topic-selector(href='#', class=`topic-${topic.UserTopic.viewed}`, data-id=topic.id, data-url=topic.url)= topic.title
```
Comment (jrob8577) [inline views/help/howtos.pug]: Can we make this a multi line attribute list for readability:
```pug
a.btn.btn-info.topic-selector(
  href='#'
  class=`topic-${topic.UserTopic.viewed}`
  data-id=topic.id
  data-url=topic.url
)= topic.title
```

File: views/help/howtos.pug [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      #commands.tab-pane.fade
+        h3 Commands
+        p Command details are coming soon!
+      #support.tab-pane.fade
```
Comment (jrob8577) [inline views/help/howtos.pug]: Please add new lines between logical blocks

File: views/items/index.pug [hunk 7]
```diff
@@ -15,6 +15,11 @@ mixin display( nodes, depth )
         +display( entry.children, depth + 1 )
 
 block content
+
+  #myModal.modal.fade(tabindex='-1', role='dialog', aria-labelledby='myModalLabel', aria-hidden='true')
```
Comment (jrob8577) [inline views/items/index.pug]: Please provide a more meaningful ID than `myModal` - maybe `helpModal`?  The classes contained in this block should also be named more meaningfully.

File: routes/help.js (hunk 4 above)
Comment (nicosesma) [inline routes/help.js]: this is the one that is over 80 chars

File: views/items/index.pug (hunk 7 above)
Comment (nicosesma) [inline views/items/index.pug]: @jrob i changed the id name and the modal is showing up but i dont know if it is rendering the correct html content

**Comment by nicosesma** [comment]
@jrob8577 made some changes according to the issues that you had pointed out, I do not know if the help functionality is working as it should though?

**Pull request closed without merging**